--- a/include/stdio.h
+++ b/include/stdio.h
@@ -22,6 +22,8 @@ __extension__
 #if defined(_FORTIFY_SOURCE) && _FORTIFY_SOURCE > 0 && defined(__OPTIMIZE__) && __OPTIMIZE__ > 0
 #include "fortify-headers.h"
 
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -118,6 +120,8 @@ _FORTIFY_FN(sprintf) int sprintf(char *_
 }
 #endif
 
+#pragma GCC diagnostic pop
+
 #endif
 
 #endif
