From ac8400ddf6aff5db873da0acfe8b183bd6692d86 Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cotequeiroz@gmail.com>
Date: Wed, 24 Jan 2024 15:59:37 -0300
Subject: [PATCH] SAE_PK: add missing obj files to sae_pk_gen

Signed-off-by: Eneas U de Queiroz <cotequeiroz@gmail.com>

Index: hostapd-2025.05.23~4b8ac10c/hostapd/Makefile
===================================================================
--- hostapd-2025.05.23~4b8ac10c.orig/hostapd/Makefile
+++ hostapd-2025.05.23~4b8ac10c/hostapd/Makefile
@@ -1256,6 +1256,7 @@ ifdef CONFIG_GETRANDOM
 CFLAGS += -DCONFIG_GETRANDOM
 endif
 OBJS += ../src/crypto/random.o
+SOBJS += ../src/crypto/random.o
 HOBJS += ../src/crypto/random.o
 HOBJS += ../src/utils/eloop.o
 HOBJS += $(SHA1OBJS)
@@ -1468,8 +1469,6 @@ SOBJS += ../src/common/sae_pk.o
 SOBJS += ../src/common/dragonfly.o
 SOBJS += $(AESOBJS)
 ifneq ($(CONFIG_TLS), mbedtls)
-SOBJS += ../src/crypto/sha384.o
-SOBJS += ../src/crypto/sha512.o
 SOBJS += ../src/crypto/sha256-prf.o
 SOBJS += ../src/crypto/sha384-prf.o
 SOBJS += ../src/crypto/sha512-prf.o
@@ -1481,7 +1480,6 @@ SOBJS += ../src/crypto/sha384-kdf.o
 SOBJS += ../src/crypto/sha512-kdf.o
 endif
 SOBJS += ../src/common/wpa_common.o
-SOBJS += ../src/crypto/random.o
 ifneq ($(CONFIG_TLS), mbedtls)
 SOBJS += ../src/crypto/sha1-prf.o
 endif
