Index: hostapd-2023-09-08-e5ccbfc6/hostapd/Makefile
===================================================================
--- hostapd-2023-09-08-e5ccbfc6.orig/hostapd/Makefile
+++ hostapd-2023-09-08-e5ccbfc6/hostapd/Makefile
@@ -16,9 +16,10 @@ endif
 ifndef LIBS_n
 LIBS_n := $(LIBS)
 endif
-ifndef LIBS_s
-LIBS_s := $(LIBS)
-endif
+# Hack to avoid linking unnecessary libs to sae_pk_gen
+#ifndef LIBS_s
+#LIBS_s := $(LIBS)
+#endif
 endif
 
 CFLAGS += $(EXTRA_CFLAGS)
