--- a/ath10k-4.13/htt_rx.c
+++ b/ath10k-4.13/htt_rx.c
@@ -1558,7 +1558,9 @@ static int ath10k_unchain_msdu(struct at
 		/* put it back together so we can free the
 		 * whole list at once.
 		 */
+#ifdef CONFIG_ATH10K_DEBUGFS
 		ar->debug.rx_drop_unchain_oom++;
+#endif
 		__skb_queue_head(amsdu, first);
 		return -1;
 	}
@@ -1595,7 +1597,9 @@ static void ath10k_htt_rx_h_unchain(stru
 	 */
 	if (decap != RX_MSDU_DECAP_RAW ||
 	    skb_queue_len(amsdu) != 1 + rxd->frag_info.ring2_more_count) {
+#ifdef CONFIG_ATH10K_DEBUGFS
 		ar->debug.rx_drop_decap_non_raw_chained++;
+#endif
 		__skb_queue_purge(amsdu);
 		return;
 	}
@@ -1613,13 +1617,17 @@ static bool ath10k_htt_rx_amsdu_allowed(
 
 	if (!rx_status->freq) {
 		ath10k_warn(ar, "no channel configured; ignoring frame(s)!\n");
+#ifdef CONFIG_ATH10K_DEBUGFS
 		ar->debug.rx_drop_no_freq++;
+#endif
 		return false;
 	}
 
 	if (test_bit(ATH10K_CAC_RUNNING, &ar->dev_flags)) {
 		ath10k_dbg(ar, ATH10K_DBG_HTT, "htt rx cac running\n");
+#ifdef CONFIG_ATH10K_DEBUGFS
 		ar->debug.rx_drop_cac_running++;
+#endif
 		return false;
 	}
 
